docs/DeveloperGuide.md

```diff
@@ -159,6 +159,50 @@ Classes used by multiple components are in the `seedu.netconnect.commons` packag
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Person Roles (Employee, Client, and Supplier)
+
+This feature update introduces three new person types into the existing Person model: `Client`, `Supplier`, and `Employee`. These classes extend the base `Person` class and encapsulate various role-specific functionalities and attributes, improving the application's ability to cater to a diverse range of user interactions.
+
+#### Overview
+
+* **Client**: Represents a customer, associated with products and preferences.
+* **Supplier**: Represents a vendor, associated with products and terms of service.
+* **Employee**: Represents an employee, associated with a department, job title, and skills.
+
+#### Implementation
+
+##### Class Additions
+
+* `Client`, `Supplier`, and `Employee` classes have been added, extending the `Person` class to include role-specific fields.
+* New classes `Department`, `JobTitle`, `Products`, `Skills`, and `TermsOfService` are introduced to encapsulate relevant attributes.
+
+##### Data Storage
+
+* The `JsonAdaptedPerson` class has been updated to support the conversion to and from the new role-based classes, ensuring compatibility with the enhanced JSON schema.
```

Comment:
similar to above, not sure about the word 'updated'

Reply:
Updated!